tempesta_fw/http_sticky.c

```diff
@@ -249,13 +259,25 @@ tfw_http_sticky_add(TfwHttpMsg *hm, u_char *value, size_t len)
 		.flags = 5 << TFW_STR_CN_SHIFT
 	};
 
-	tfw_http_prep_hexstring(buf, value, len);
+	/*
+	 * Asynchronous access to conn->peer of a client connection.
+	 * A connection may go away at any time, so it's necessary
+	 * to protect that with a lock. If the client connection
+	 * is already gone, then there's nothing we can do here.
+	 */
+	spin_lock(&hmreq->conn->splock);
+	client = (TfwClient *)hmreq->conn->peer;
+	if (client)
+		tfw_http_prep_hexstring(buf, client->cookie.hmac, len);
+	spin_unlock(&hmreq->conn->splock);
+	if (client == NULL)
+		return -ENOENT;
```

Comment:
Which data exactly and where can be concurrently accessed that we must protect it with the lock?
There is also client value double check.